src/inmanta/ast/statements/generator.py

```diff
@@ -315,8 +342,33 @@ def execute(self, requires: Dict[object, object], resolver: Resolver, queue: Que
         # the type to construct
         type_class = self.type.get_entity()
 
+        # kwargs
+        direct_kwarg_attrs: Dict[str, InmantaType] = {}
+        indirect_kwarg_attrs: Dict[str, ExpressionStatement] = {}
+        for kwargs in self.wrapped_kwargs:
+            for (k, v) in kwargs.execute(requires, resolver, queue):
+                if k in self.attributes or k in direct_kwarg_attrs or k in indirect_kwarg_attrs:
+                    raise RuntimeException(
+                        self, "The attribute %s in the constructor call of %s is already set." % (k, self.class_type)
+                    )
+                if k in self.required_kwargs:
+                    direct_kwarg_attrs[k] = v
+                else:
+                    attribute = self.type.get_entity().get_attribute(k)
```

Comment:
this part doesn't make sense to me.

Why make an expression when the value is known.
Adding them to the indirect attributes adds overhead but no value.

Indirection is used to defer execution (i.e. make requires_emit return a smaller set of requirements), in this case, all values are known,  requires_emit's size has not been reduced.

Reply:
This is what we talked about Wednesday. We must've had a misunderstanding. I actually started out by just adding them to the instance instantiation. From our discussion I gathered that that wasn't a good idea (something to do with list freezes) and I assumed you meant they should be deferred instead.

Reply:
yes, they should, but here they aren't

Reply:
cleaned up for now, see #1711 and #1712 for more efficient approaches.